transport/impls/ws/client.ts

```diff
@@ -44,14 +45,24 @@ export class WebSocketClientTransport extends ClientTransport<WebSocketConnectio
         connectedTo: to,
       });
 
-      this.wsGetter(to)
+      const urlMaybePromise = this.urlGetter(to);
+      const urlPromise =
+        urlMaybePromise instanceof Promise
+          ? urlMaybePromise
+          : Promise.resolve(urlMaybePromise);
+
+      urlPromise
```

Comment:
```suggestion
      Promise.resolve(this.urlGetter(to))
```

I think this should work